src/core/src/pass/constant_folding.cpp

```diff
@@ -105,6 +105,24 @@ bool ov::pass::ConstantFolding::run_on_model(const std::shared_ptr<ov::Model>& m
 
     for (const auto& original_node : model->get_ordered_ops()) {
         auto node = original_node;
+        if (!original_node->can_constant_fold(original_node->input_values())) {
+            if (auto sub_graph_node = std::dynamic_pointer_cast<ov::op::util::MultiSubGraphOp>(node)) {
+                // recursively constant fold operators containing subgraphs (ie: TensorIterator, Loop)
+                size_t sub_graphs_num = sub_graph_node->get_internal_subgraphs_size();
+                for (size_t sub_graph_ind = 0; sub_graph_ind < sub_graphs_num; ++sub_graph_ind) {
+                    rewritten =
+                        run_on_model(sub_graph_node->get_function(static_cast<int>(sub_graph_ind))) || rewritten;
+                }
+            }
+
+            // if CF was unsuccessful remove original precision attribute from inputs
+            bool restored = restore_original_input_precision(original_node);
```

Comment:
we don't need to restore the precision here because now we replace it later in lines 127-129.
I believe we need to leave it as it was in the "else" branch
<img width="325" alt="{68329826-E636-42E1-A13E-D848860B2D09}" src="https://github.com/user-attachments/assets/e4b42356-f4f4-43ab-94ac-55d576b4dead">

Reply:
We don't reach precision restore later since there is `continue` in this branch.
The test `constant_folding/UnsupportedTypesTest.add_multiply` fails because precision restore is not done for `Multiply` node. So, I added `restore_original_input_precision` call in front of `continue`.
Please let me know if this is OK or not.

Reply:
With `restore_original_input_precision` call in front of `continue` all tests are passed expect PyTorch test `TestLoopWithAlias::test_loop_alias` with the error
```
E       FrontEnd API failed with OpConversionFailure:
E       [PyTorch Frontend] Translation for prim::ListConstruct support only constant inputs
E       
E       Summary:
E       -- normalize step failed with: Check 'convert->constant_fold(replacements, convert->input_values())' failed at src/core/src/pass/constant_folding.cpp:68
```
The error happens because `all_constants`=0 in `Node::can_constant_fold` called by `Node::constant_fold` called by restore_original_input_precision with `ConvertPromoteTypes`.
`ConvertPromoteTypes` has no `constant_fold` overriden, so `can_constant_fold` is not overriden as well.
Not sure what the issue is here.